src/Instance/MoveHoldingContext/MoveHoldingContext.test.js

```diff
@@ -0,0 +1,225 @@
+
+    expect(getByText(/ConfirmationModal/)).toBeInTheDocument();
+    expect(getByText(/0 items will be moved to/)).toBeInTheDocument();
+  });
+
+  it('should render correct list of holdings for Annex with checkbox', async () => {
+    const { getByTestId } = renderMoveHoldingContext();
+
+    const annexSection = getByTestId('c4a15834-0184-4a6f-9c0c-0ca5bad8286d');
+    const annexHoldingsAccordionBtn = getByTestId('accordion-toggle-button-c4a15834-0184-4a6f-9c0c-0ca5bad8286d');
+
+    fireEvent.click(annexHoldingsAccordionBtn);
+
+    let grid;
+    let rows;
+
+    await waitFor(() => {
+      grid = within(annexSection).getByRole('grid');
+      rows = within(grid).getAllByRole('row');
+
+      expect(grid).toBeVisible();
+      expect(rows).toHaveLength(3);
+
+      // render correct table headers
+      const titles = [
+        'Item: barcode',
+        'Status',
+        'Copy number',
+        'Loan type',
+        'Effective location',
+        'Enumeration',
+        'Chronology',
+        'Volume',
+        'Year, caption',
+        'Material type'
+      ];
+
+      titles.forEach(title => {
+        expect(within(rows[0]).getByText(title)).toBeInTheDocument();
+      });
+
+      // second row should have 'Inactive' cell
+      expect(within(rows[1]).getByText('Inactive')).toBeInTheDocument();
+
+      // should render 'End of list' indicator
+      expect(within(grid).getByText('End of list')).toBeInTheDocument();
+    });
+
+    fireEvent.click(within(rows[1]).getByRole('checkbox'));
+
+    rows.forEach(row => {
+      expect(within(row).getByRole('checkbox')).toBeChecked();
+    });
+
+    fireEvent.click(within(rows[1]).getByRole('checkbox'));
+
+    rows.forEach(row => {
+      expect(within(row).getByRole('checkbox')).not.toBeChecked();
+    });
+  });
+
+  describe('when "Move" button is clicked', () => {
+    const clickMoveFlow = ({ getByTestId, getByText }) => {
+      const holdingsAnnex = getByTestId('item-row-c4a15834-0184-4a6f-9c0c-0ca5bad8286d');
+
+      const moveToBtn = within(holdingsAnnex).getByText('Move to');
+
+      fireEvent.click(moveToBtn);
+
+      const dropdownMoveBtn = within(holdingsAnnex).getByText('A journey through Europe Bildtontraeger high-speed lines European Commission, Directorate-General for Mobility and Transport');
+      expect(dropdownMoveBtn).toBeInTheDocument();
+
+      fireEvent.click(dropdownMoveBtn);
+
+      expect(getByText('1 holding will be moved to')).toBeInTheDocument();
+    };
+
+    it('should move selected holdings if "Confirm button" is clicked', async () => {
+      const { getByText, getByTestId } = renderMoveHoldingContext();
+
+      clickMoveFlow({ getByText, getByTestId });
+
+      const confirmBtn = screen.getByRole('button', { name: /confirm/ });
+
+      fireEvent.click(confirmBtn);
+
+      expect(screen.queryByText('Loading')).toBeInTheDocument();
+    });
+
+    it('should close modal and stop moving when "Cancel" is clicked', async () => {
```

Comment:
there are some extra `async` declarations left, also [here](https://github.com/folio-org/ui-inventory/pull/2469/files#diff-054a3ed1ef8fb650f861d0d464f2632a9911ee37779b23f3f4db12d096321ae9R122), and [here](https://github.com/folio-org/ui-inventory/pull/2469/files#diff-054a3ed1ef8fb650f861d0d464f2632a9911ee37779b23f3f4db12d096321ae9R201)

Reply:
I see, removed. My prev solution contained a lot of async data, but then I was able to avoid it, but I forgot remove all these async indicators